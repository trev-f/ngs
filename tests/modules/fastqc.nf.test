nextflow_process {

    name "Test Process FASTQC"
    script "modules/fastqc.nf"
    process "FASTQC"

    test("FASTQC works for single-end reads.") {

        when {
            params {}
            process {
                """
                input[0] = [
                    [sampleName: "SRR1066657_GSM1299413_WT_NR_A"],
                    [
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz")
                    ]
                ]
                """
            }
        }

        then {
            assert process.success

            with(process.out.html) {
                assert size() == 1
                assert get(0) ==~ /^.*\/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000_fastqc.html/
            }
            with(process.out.zip) {
                assert size() == 1
                assert get(0) ==~ /^.*\/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000_fastqc.zip/
            }

            // We can only check snapshot for fastqc html report bc MD5 hash is different for zip files produced at different times.
            assert snapshot(process.out.html).match()
        }

    }

    test("FASTQC works for paired-end reads.") {

        when {
            params {}
            process {
                """
                input[0] = [
                    [sampleName: "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1"],
                    [
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000.fastq.gz"),
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000.fastq.gz")
                    ]
                ]
                """
            }
        }

        then {
            assert process.success

            with(process.out.html.get(0)) {
                assert size() == 2
                assert get(0) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000_fastqc.html/
                assert get(1) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000_fastqc.html/
            }
            with(process.out.zip.get(0)) {
                assert size() == 2
                assert get(0) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000_fastqc.zip/
                assert get(1) ==~ /^.*\/SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000_fastqc.zip/
            }

            // We can only check snapshot for fastqc html report bc MD5 hash is different for zip files produced at different times.
            assert snapshot(process.out.html).match()
        }

    }

}
