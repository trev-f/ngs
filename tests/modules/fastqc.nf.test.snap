{
    "FASTQC works for paired-end reads.": {
        "content": [
            [
                [
                    "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_1_50000_fastqc.html:md5,b0678b630b35c72e417c86b03b3e9391",
                    "SRR6924589_GSM3073211_Saccharomyces_cerevisiae-AN_Biological_Repeat-1_Saccharomyces_cerevisiae_RNA-Seq_2_50000_fastqc.html:md5,d5997e0fb28d7a69372b926340241df4"
                ]
            ]
        ],
        "timestamp": "2023-07-25T16:06:41+0000"
    },
    "FASTQC works for single-end reads.": {
        "content": [
            [
                "SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000_fastqc.html:md5,910f4febdc64161348848a7ee055c40a"
            ]
        ],
        "timestamp": "2023-07-25T16:06:41+0000"
    }
}